src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/CommonExtensions.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="CommonExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Instrumentation.AWSLambda.Implementation
+{
+    internal static class CommonExtensions
+    {
+        internal static bool TryGetValueIgnoringCase<T>(this IDictionary<string, T> dict, string key, out T value)
+        {
+            value = default;
+            var foundTargetKey = false;
+            foreach (var kvp in dict)
+            {
+                if (string.Equals(kvp.Key, key, StringComparison.OrdinalIgnoreCase))
+                {
+                    value = kvp.Value;
+                    foundTargetKey = true;
+                    break;
+                }
+            }
+
+            return foundTargetKey;
```

Comment:
```suggestion
            foreach (var kvp in dict)
            {
                if (string.Equals(kvp.Key, key, StringComparison.OrdinalIgnoreCase))
                {
                    value = kvp.Value;
                    return true;
                }
            }

            value = default;
            return false;
```

Adding boolean control variables makes the code much harder to read IMHO.

Reply:
I was thinking to keep only the `value` variable and compare with the `default`, but what if value stored in `dict` is equal to default ?

Reply:
`default` is `null` here, isn't it? So that would work as well.

Reply:
This method has been removed.

Reply:
I think this conversation can be resolved as it became obsolete